File: packages/schemas/src/phase-agent-profile.ts
/** Phase Agent Profile types matching backend schemas. */

export type WorkflowPhase =
  | "specify"
  | "plan"
  | "tasks"
  | "implement"
  | "review"
  | "validate";

export interface PhaseAgentProfile {
  id: string;
  project_id: string;
  phase: WorkflowPhase;
  agent_id: string;
  priority: number;
  is_default: boolean;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface PhaseAgentProfileCreate {
  phase: WorkflowPhase;
  agent_id: string;
  priority?: number;
  is_default?: boolean;
  notes?: string | null;
}

export interface PhaseAgentProfileList {
  items: PhaseAgentProfile[];
  total: number;
}

File: apps/api/app/schemas/phase_agent_profile.py
"""FM-111: Phase Agent Profile schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.phase_agent_profile import WorkflowPhase


class PhaseAgentProfileCreate(BaseModel):
    phase: WorkflowPhase
    agent_id: uuid.UUID
    priority: int = 0
    is_default: bool = False
    notes: str | None = None


class PhaseAgentProfileUpdate(BaseModel):
    agent_id: uuid.UUID | None = None
    priority: int | None = None
    is_default: bool | None = None
    notes: str | None = None


class PhaseAgentProfileRead(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    phase: WorkflowPhase
    agent_id: uuid.UUID
    priority: int
    is_default: bool
    notes: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PhaseAgentProfileList(BaseModel):
    items: list[PhaseAgentProfileRead]
    total: int
